packages/block-library/src/navigation-link/index.php

```diff
@@ -120,6 +120,40 @@ function block_core_navigation_link_render_submenu_icon() {
 	return '<svg xmlns="http://www.w3.org/2000/svg" width="12" height="12" viewBox="0 0 12 12" fill="none" aria-hidden="true" focusable="false"><path d="M1.50002 4L6.00002 8L10.5 4" stroke-width="1.5"></path></svg>';
 }
 
+/**
+ * Checks if the given url is encoded
+ *
+ * @param string $url The url to check.
+ *
+ * @return boolean Whether or not a url is encoded.
+ */
+function is_url_encoded($url) {
+    $query = parse_url($url, PHP_URL_QUERY);
+    $query_params = wp_parse_args($query);
+    foreach($query_params as $query_param){
+        if(rawurldecode($query_param) !== $query_param){
+            return true;
+        }
+    }
+    return false;
+}
+
+
+/**
+ * Decodes a url if it's encoded, returning the same url if not.
+ *
+ * @param string $url The url to decode.
+ *
+ * @return string $url Returns the decoded url.
+ */
+function urldecode_once($url){
+    if(is_url_encoded($url)){
+        return rawurldecode($url);
+    }
+    return $url;
+}
```

Comment:
Can we consolidate these two functions into one function called `block_core_navigation_link_maybe_urldecode()`?

For better or for worse, `maybe_*` functions are a common pattern in WordPress core: https://developer.wordpress.org/?s=maybe

It seems like there are some PHPCS issues that need to be fixed too.

Reply:
I prefer having them as two separate functions, as it's clear what each one of them is doing, and we can reuse `is_url_encoded` for testing encoded URLs elsewhere if we need to instead of tying those two functions together and producing a "side effect" if the URL is encoded.

However, I'll rename the `is_url_encoded` to `maybe_urlencoded` to match the common pattern in WordPress core.

Thanks for pointing this out!

Reply:
@kozer Given this will eventually be included in WordPress core, I'd like them consolidated to one function to match the pattern already established in WordPress core.

Reply:
@danielbachhuber Doesn't the `maybe_urldecode` renaming establish that pattern?

Reply:
@kozer If it's only one `block_core_navigation_link_maybe_urldecode()` function, then yes 😊

In the context of a large open source project, we shouldn't introduce two functions if we only need one.

Reply:
@danielbachhuber I made the change as requested! 
A side note:
I'm not sure if the phpcs errors you mentioned are related to tabs vs spaces. Neither vscode or neovim show any warnings/errors to me.
Are tabs the way to go in PHP-related files?

Reply:
> I'm not sure if the phpcs errors you mentioned are related to tabs vs spaces. Neither vscode or neovim show any warnings/errors to me.
> Are tabs the way to go in PHP-related files?

@kozer Here's the failing test: https://github.com/WordPress/gutenberg/actions/runs/3830602130/jobs/6518686126

The failing test runs `npm run lint:php`:

<img width="699" alt="image" src="https://user-images.githubusercontent.com/36432/210393394-e9e2abe4-aeb8-428e-ab59-28cd65822d78.png">

You can run `npm run lint:php` in your local environment to see all of the same errors.

I use this VS Code extension to integrate PHPCS:

Name: PHP Sniffer & Beautifier
Id: valeryanm.vscode-phpsab
Description: PHP Sniffer & Beautifier for Visual Studio Code
Version: 0.0.15
Publisher: Samuel Hilson
VS Marketplace Link: https://marketplace.visualstudio.com/items?itemName=ValeryanM.vscode-phpsab

You can see all of WordPress' PHP coding standards here: https://developer.wordpress.org/coding-standards/wordpress-coding-standards/php/